numba/tests/test_dyn_array.py

```diff
@@ -854,6 +854,67 @@ def func(n, m, k):
         self.check_eye_n_m_k(func)
 
 
+class TestNdDiag(TestCase):
+
+    def setUp(self):
+        v = np.array([1, 2, 3])
+        hv = np.array([[1, 2, 3]])
+        vv = np.transpose(hv)
+        self.vectors = [v, hv, vv]
+        a3x4 = np.arange(12).reshape(3, 4)
+        a4x3 = np.arange(12).reshape(4, 3)
+        self.matricies = [a3x4, a4x3]
+        def func(q):
+            return np.diag(q)
+        self.f_diag = func
+
+        def func_kwarg(q, k=0):
+            return np.diag(q, k=k)
+        self.f_diag_kwarg = func_kwarg
+
+    def check_diag(self, pyfunc, *args, **kwargs):
+        cfunc = nrtjit(pyfunc)
```

Comment:
It's probably a bit wasteful to recompile pyfunc every time check_diag() is called.